src/core.cpp

```diff
@@ -61,22 +118,60 @@ CVC::CVC(
         std::unique_ptr<DecisionEngine> decision_engine,
         std::vector<std::unique_ptr<Character>> characters,
         std::mt19937 random_generator) {
+    this->ticks_ = 0;
     this->random_generator_ = random_generator;
     this->decision_engine_ = std::move(decision_engine);
     this->characters_ = std::move(characters);
 }
 
+std::vector<Character *> CVC::GetCharacters() const {
+    std::vector<Character *> ret;//(this->characters_.size());
+    for(auto& character : this->characters_) {
+        ret.push_back(character.get());
+    }
+    return ret;
+}
+
 void CVC::GameLoop() {
     //TODO: don't just loop for some random hardcoded number of iterations
-    for (int i = 0; i < 10; i++) {
+    for (; this->ticks_ < 1000; this->ticks_++) {
+        //0. expire relationships
+        this->ExpireRelationships();
         //1. evaluate queued actions
         this->EvaluateQueuedActions();
         //2. choose actions for characters
         this->ChooseActions();
 
         //TODO: expose state somehow to keep track of how stuff is going
         //  maybe also use the record of the state as training
-        printf(".");
+        this->PrintState();
+    }
+}
+
+void CVC::PrintState() const {
+    //TODO: probably don't want to always spit to stdout
+
+    printf("tick %d:\n", this->ticks_);
+    for(const auto& character: this->characters_) {
```

Comment:
nit: You're inconsistent with spacing between if/for and the opening paren. I think it is most common to always have a space there.

Reply:
I'll check out clang-format which should probably help here